common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -600,35 +583,19 @@ private boolean runShutdownHooks() {
     @Override
     @Deprecated
     public void shutdown() {
-        if (isShutdown()) {
+        if (isShuttingDown()) {
```

Comment:
I think you just moved this check from line 611 ... however I wonder if the intention is that this method should be able to "force the shutdown" even if the state is `ST_SHUTTING_DOWN` (@normanmaurer - WDYT). If that is the case this method should only exit prematurely if `state >= ST_SHUTDOWN`.